chain/network/src/peer_manager/peer_store.rs

```diff
@@ -96,7 +101,12 @@ impl PeerStore {
         for (peer_id, peer_state) in store.list_peer_states()? {
             // If it’s already banned, keep it banned.  Otherwise, it’s not connected.
             let status = if peer_state.status.is_banned() {
-                peer_state.status
+                if connect_only_to_boot_nodes && boot_nodes_set.contains(&peer_id) {
+                    // Give boot node another chance.
```

Comment:
wouldn't it be easier to early exit at the start of the constructor in case connect_only_to_boot_nodes is true?
Or have a dedicated constructor of PeerStore?